### DISCUSSION TITLE
Missing borders on floating windows

### DISCUSSION THREAD

**Discussion by aryklein** [VonHeikemen/lsp-zero.nvim#217] [Q&A] [answered]
I just upgraded to version 2.x and I lost borders (and other configs) on floating windows.

This is my lsp-zero config:

```lua
local lsp = require('lsp-zero').preset('recommended')

lsp.on_attach(function(client, bufnr)
    lsp.default_keymaps({buffer = bufnr})
end)

require('lspconfig').lua_ls.setup(lsp.nvim_lua_ls())

lsp.setup_nvim_cmp({
    formatting = {
        fields = {'abbr', 'kind', 'menu'},
        format = require('lspkind').cmp_format({
            mode = 'symbol_text', -- show only symbol annotations
            maxwidth = 50, -- prevent the popup from showing more than provided characters (e.g 50 will not show more than 50 characters)
            ellipsis_char = '...', -- when popup menu exceed maxwidth, the truncated part would show ellipsis_char instead (must define maxwidth first)
        })
    }
})

lsp.setup()
```

I also tried by adding:
```lua
-- diagnostic config
vim.diagnostic.config({
    virtual_text = true,
    signs = true,
    update_in_insert = false,
    underline = true,
    severity_sort = false,
    float = {
        focusable = false,
        style = 'minimal',
        border = 'rounded',
        source = 'always',
        header = '',
        prefix = '',
    },
})
```
but didn't work.

This is how it looks now:

![2023-04-25_02-03-17](https://user-images.githubusercontent.com/8699850/234180467-c13cdb9b-465d-4ef8-91da-52e2c0b92559.png)

Do you know how to restore diagnostics borders?

**Comment by VonHeikemen** [comment]
Use the `cmp` module to configure the completion menu.

```lua
local lsp = require('lsp-zero').preset('recommended')

require('lspconfig').lua_ls.setup(lsp.nvim_lua_ls())

lsp.setup()

local cmp = require('cmp')

cmp.setup({
  window = {
      completion = cmp.config.window.bordered(),
      documentation = cmp.config.window.bordered(),
  },
  formatting = {
      fields = {'abbr', 'kind', 'menu'},
      format = require('lspkind').cmp_format({
          mode = 'symbol_text', -- show only symbol annotations
          maxwidth = 50, -- prevent the popup from showing more than provided characters (e.g 50 will not show more than 50 characters)
          ellipsis_char = '...', -- when popup menu exceed maxwidth, the truncated part would show ellipsis_char instead (must define maxwidth first)
      })
  }
})
```

side note: you don't need to call `default_keymaps()` when using the `recommended` preset.

**Reply by aryklein** [reply]
Thank you so much, @VonHeikemen. Your solution solved my issue. However, I was wondering if the `preset('recommended')` option should have been applied it by default. Could you please clarify this for me?

I found more information on this in the following links:

https://github.com/VonHeikemen/lsp-zero.nvim/blob/v2.x/doc/md/lsp.md#diagnostics
https://github.com/VonHeikemen/lsp-zero.nvim/blob/v2.x/doc/md/api-reference.md#recommended

I have also attached a screenshot:

![2023-04-25_10-20-23](https://user-images.githubusercontent.com/8699850/234290889-bdfa111c-7fcd-4665-b867-3848c1a5a2fa.png)

Alternatively, I used the `vim.diagnostic.config()` function in version 1.x."

**Reply by VonHeikemen** [reply, marked as answer]
All the options for the completion menu are under `manage_nvim_cmp`. The option `float_border` modifies the windows for the "hover" action, signature help, and error message window.
